File: utbot-framework/src/main/java/org/utbot/quickcheck/generator/java/util/SetGenerator.java


package org.utbot.quickcheck.generator.java.util;

import org.utbot.quickcheck.generator.Size;
import org.utbot.quickcheck.generator.java.util.CollectionGenerator;

import java.util.Set;

/**
 * Base class for generators of {@link Set}s.
 *
 * @param <T> the type of set generated
 */
public abstract class SetGenerator<T extends Set>
    extends CollectionGenerator<T> {

    protected SetGenerator(Class<T> type) {
        super(type);
    }

    @Override public void configure(Size size) {
        super.configure(size);

        setDistinct(true);
    }
}

File: utbot-framework/src/main/kotlin/org/utbot/quickcheck/generator/java/util/SetGenerator.kt
package org.utbot.quickcheck.generator.java.util

import org.utbot.quickcheck.generator.Size

/**
 * Base class for generators of [Set]s.
 *
 * */
abstract class SetGenerator constructor(type: Class<*>) : CollectionGenerator(type) {
    override fun configure(size: Size) {
        super.configure(size)
        setDistinct(true)
    }
}